File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/token/RefreshTokenRepository.java
package br.com.deltaglobalbank.identity.domain.token;

import java.util.List;
import java.util.UUID;

public interface RefreshTokenRepository {
    RefreshToken findById(UUID id);

    RefreshToken findByTokenHash(String hash);

    List<RefreshToken> findAllByUserId(UUID userId);

    List<RefreshToken> findAllActiveByUserId(UUID userId);

    RefreshToken save(RefreshToken refreshToken);

    List<RefreshToken> saveAll(List<RefreshToken> refreshTokens);
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/token/RefreshTokenRepository.kt
package br.com.deltaglobalbank.identity.domain.token

import java.util.UUID

interface RefreshTokenRepository {
    fun findById(id: UUID): RefreshToken?
    fun findByTokenHash(hash: String): RefreshToken?
    fun findAllByUserId(userId: UUID): List<RefreshToken>
    fun findAllActiveByUserId(userId: UUID): List<RefreshToken>
    fun save(refreshToken: RefreshToken): RefreshToken
    fun saveAll(refreshTokens: List<RefreshToken>): List<RefreshToken>
}
